### DISCUSSION TITLE
TypeError: connector.connect is not a function

### DISCUSSION THREAD

**Discussion by tigerraj32** [WalletConnect/walletconnect-monorepo#1363] [Developer Support]
I am using the following hooks but somehow connect() is not recognized and getting `TypeError: connector.connect is not a function`

` const connector = useWalletConnect();

connector.connect()
`

I have @walletconnect/react-native-dapp: 1.7.0 installed.

**Comment by BenRacicot** [comment]
I'm not a WC engineer but the init and connect methods are asynchronous.

I've converted mine to RxJS observables but you'll need to use async/await and could also work with them as `promises` with `.then`

**Reply by hackyguru** [reply]
+1, thanks for helping the community @BenRacicot
